src/ui/auth/rackspace/rackspace.go

```diff
@@ -0,0 +1,174 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package rackspace
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/vmware/harbor/src/common/dao"
+	"github.com/vmware/harbor/src/common/models"
+	"github.com/vmware/harbor/src/common/utils/log"
+	"github.com/vmware/harbor/src/ui/auth"
+	"github.com/vmware/harbor/src/ui/config"
+)
+
+// Auth implements Authenticator interface to authenticate against kubernetes-auth
+type Auth struct{}
+
+// AuthResponse is the response from Rackspace Managed Kubernetes Auth (kubernetes-auth)
+// If the user is authenticated, the User struct is filled. Otherwise, the Error string is filled
+type AuthResponse struct {
+	APIVersion string `json:"apiVersion"`
+	Kind       string `json:"kind"`
+	Status     struct {
+		Authenticated bool   `json:"authenticated"`
+		Error         string `json:"error,omitempty"`
+		User          struct {
+			Username string              `json:"username"`
+			UID      string              `json:"uid,omitempty"`
+			Groups   []string            `json:"groups,omitempty"`
+			Extra    map[string][]string `json:"extra,omitempty"`
+		} `json:"user,omitempty"`
+	} `json:"status"`
+}
+
+var (
+	rackspaceMK8SAuthURLTokenEndpoint string
+)
+
+// Authenticate checks user's credential against the Rackspace Managed Kubernetes Auth (kubernetes-auth)
+// if the check is successful a dummy record will be inserted into DB, such that this user can
+// be associated to other entities in the system.
+func (l *Auth) Authenticate(m models.AuthModel) (*models.User, error) {
+
+	// kubernetes-auth only uses the token (m.Password) for auth. The username (m.Principal) isn't used at all.
+	// In fact, a user could put anything at all into the username field. It must be ignored.
+	// However, we log the username to help track the request because we can't put the token in the logs.
+	log.Debugf("ProvidedUsername=%s Authentication attempt", m.Principal)
+
+	// build auth request
+	authRequestBody := fmt.Sprintf("{\"spec\": {\"token\": \"%s\"}}", m.Password)
+	req, err := http.NewRequest(http.MethodPost, rackspaceMK8SAuthURLTokenEndpoint, strings.NewReader(authRequestBody))
+	if err != nil {
+		log.Errorf("ProvidedUsername=%s Error=%v", m.Principal, err)
+		return nil, err
+	}
+
+	log.Debugf("ProvidedUsername=%s Sending auth request to %s", m.Principal, rackspaceMK8SAuthURLTokenEndpoint)
+
+	// send auth request
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
```

Comment:
this seems to use a lot of code. how do you feel replacing lines 66-77 with a one liner:

```
resp, err := http.Post(rackspaceMK8SAuthURLTokenEndpoint, "application/json", bytes.NewBuffer(jsonVal))
```